docs/edgecases.md

```diff
@@ -1,49 +1,66 @@
 # Edge cases
+
 ## Preamble
-The python ecosystem has a very long history. Over the years, an uncountable number of modules have been created, but more importantly, the assembly technology of these modules has changed many times. `ez_install`, `setuptools`, `distutils`, eggs, `pip`, wheels, PEP-518, PEP-517, `flit`, `poetry`. All this diversity creates great difficulties. Poetry, on the one hand, being a build system, can abandon the setuptools and distutils formats, but at the same time, being a dependency installation and management system, it must support all current distribution technologies. And it can do this without revealing the internal structure and the machinery behind curtains. In other words, when you use Poetry, both `setuptools`, `pip` and other build tools can be used to install some modules, and this happens without formally declaring dependence on these tools.
 
-All these factors can cause the module not to be assembled or installed in a certain environment. Nix is a much stricter ecosystem, it has more formal requirements, but at the same time more guarantees. And since `poetry2nix` relies on Nix and its strict rules in its work, there are many rough edges and inaccuracies in the metadata on pypi.org leads to errors.
-This is very sad, and we will inevitably face the fact that some modules can not be installed, can not be built, or just cause errors, have unnecessary or missing direct and transitive dependencies. However, Nix is a very flexible system that allows you to modify some aspects of the module with pinpoint accuracy, without changing the source code of the module (and sometimes changing it). This is what this section is dedicated to. We have tried to collect here and describe typical errors and ways to eliminate them.
+The Python ecosystem has a very long history.
```

Comment:
Please get rid of the spurious newline changes, they make things difficult to review.

Reply:
Actually it makes it easier to see changes if you put each sentence on a separate line.
That's why I did it as the first commit so that it's clear what I changed afterwards.
In the whole "Files changed" section that might not be so clear but if you look at it commit by commit it makes more sense and future changes will be more obvious.
However, I can still change it if you wish.

Reply:
GitHub handles inline diffs just fine, so I don't think this is necessary.

Reply:
New lines are removed.